File: MediaLibrary/MediaLibraryWebUI/lib/app/configurations/music-configuration.ts
import BaseConfiguration from './base-configuration'
import * as Enums from '../assets/enums'

export default class MusicConfiguration extends BaseConfiguration {
    private selectedAlbumId: number;
    private selectedArtistId: number;
    private selectedAlbumSort: Enums.AlbumSort;
    private selectedArtistSort: Enums.ArtistSort;
    private selectedSongSort: Enums.SongSort;
    private selectedMusicTab: Enums.MusicTabs;
    private selectedMusicPage: Enums.MusicPages;

    constructor() {
        super();
    }
}

File: ML/ML_WebUI/Models/Configurations/MusicConfiguration.cs
﻿using MediaLibraryDAL.DbContexts;
using MediaLibraryWebUI.Models.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using static MediaLibraryWebUI.UIEnums;

namespace MediaLibraryWebUI.Models.Configurations
{
    public class MusicConfiguration : BaseConfiguration
    {
        public MusicConfiguration()
        {
        }

        public int SelectedAlbumId { get; set; }
        public int SelectedArtistId { get; set; }
        public AlbumSort SelectedAlbumSort { get; set; }
        public ArtistSort SelectedArtistSort { get; set; }
        public SongSort SelectedSongSort { get; set; }
        public MusicTabs SelectedMusicTab { get; set; }
        public MusicPages SelectedMusicPage { get; set; }
        public string PreviousSearchQuery { get; set; }
    }
}